src/main/plugins/websearch-plugin/websearch-plugin.ts

```diff
@@ -95,24 +95,32 @@ export class WebSearchPlugin implements ExecutionPlugin {
         });
     }
 
-    private getSearchTerm(webSearchEngine: WebSearchEngine, userInput: string): string {
+    private buildDescriptionFromUserInput(webSearchEngine: WebSearchEngine, userInput: string): string {
+        return this.buildDescriptionFromSearchTerm(webSearchEngine, this.getSearchTerm(webSearchEngine, userInput, true));
+    }
+
+    private buildDescriptionFromSearchTerm(webSearchEngine: WebSearchEngine, searchTerm: string): string {
+        return this.translationSet.websearchDescription
+            .replace("{{websearch_engine}}", webSearchEngine.name)
+            .replace("{{search_term}}", searchTerm);
+    }
+
+    private getSearchTerm(webSearchEngine: WebSearchEngine, userInput: string, doNotEncode = false): string {
         let searchTerm = userInput.replace(webSearchEngine.prefix, "");
 
-        if (webSearchEngine.encodeSearchTerm) {
+        if (webSearchEngine.encodeSearchTerm && !doNotEncode) {
```

Comment:
`!doNotEncode` is in my opinion a little overcomplicated. Not not conditions are quite hard to read, so I suggest to rename this variable to `skipEncoding` or something similar. What do you think?

Reply:
You are right, skipEnoding sounds better.